src/odinapi/views/level2.py

```diff
@@ -1144,10 +1160,23 @@ def _fetch(self, version, project):
             param = parse_parameters()
         except ValueError as e:
             raise BadRequest(str(e))
+
         db = level2db.Level2DB(project)
+        products = param['products']
         meas_iter = db.get_measurements(param.pop('products'), **param)
-        # TODO: Limit/paging
-        return meas_iter
+        if version == 'v4':
+            return meas_iter
+        scans = level2db.get_valid_collapsed_products(
+            list(meas_iter), get_args.get_int('document_limit'))
```

Comment:
shouldn't the limit be the limit used in `get_measurements`? That limit is actually set inside the function since it has a default value. So either it must not have a default value and be decided here, or the limit used in the function must be returned from the function.

Reply:
now corrected.